### PR TITLE
Logistics items (BUY ITEM PART 2)

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:

DON'T MERGE THIS BEFORE THE BUY ITEM GUI (<- For Bob or Frost)

    This is part 2 buy item gui. It adds pack-able objects and more objects to buy.
    Repair stations can be bought for repairing.
    Vehicle ammo crates can be bought for rearming.
    Tired of having to load out your car with medical items, well now you don't have to. You can buy a medical box full of medical items, depending on whether or ace is on.
    Medical tents were added and the main motivation for packable objects. Place a medical tent down and start healing your allies in a mass cas.
    

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Comment by killerswin2** [comment]
overview

**Comment by Bob-Murphy** [comment]
@killerswin2 from what I see this is good to mess with and review, right?

**Comment by killerswin2** [comment]
@Bob-Murphy yes

**Comment by Bob-Murphy** [comment]
Unpacking a large asset just plops it somewhere, not checking the environment

**Comment by Bob-Murphy** [comment]
You should not be able to carry around the massive thingies once they are unpacked
- medical tent
- repair station

**Comment by Bob-Murphy** [comment]
needs implementation into templates so we can set other assets for example for SOG PF

**Comment by killerswin2** [comment]
> You should not be able to carry around the massive thingies once they are unpacked
> 
>     * medical tent
> 
>     * repair station

That is allowed because packing and unpacking again to move the objects two more meters is detrimental to players.

File: A3A/addons/core/Templates/Templates/FactionDefaults/RebelDefaults.sqf [hunk 1]
```diff
@@ -13,8 +13,19 @@
 ["flyGear", ["U_I_pilotCoveralls"]] call _fnc_saveToTemplate;
 
 ["vehicleLightSource", "Land_LampShabby_F"] call _fnc_saveToTemplate;
-["vehicleFuelDrum", ["FlexibleTank_01_forest_F", 160]] call _fnc_saveToTemplate;
-["vehicleFuelTank", ["B_Slingload_01_Fuel_F", 1000]] call _fnc_saveToTemplate;
+["vehicleFuelDrum", ["FlexibleTank_01_forest_F", 50]] call _fnc_saveToTemplate;
+["vehicleFuelTank", ["B_Slingload_01_Fuel_F", 500]] call _fnc_saveToTemplate;
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/FactionDefaults/RebelDefaults.sqf:17]: Fuel is fine, you constantly have Access to it/ You dont have to drive far to get it anyway.

I dont think 500 is a good price for the rest of the Items, we either increase the Prices or make them accessible at higher Warlevels


Repair, well with Vanilla its unlimited Repairs anywhere anytime, almost instantly, can lead to massive cheese if you carry it around everywhere and repair in Combat, With Ace it's not as Useful as you need to Manually repair/Perform the Full Repair (Depending on Settings etc.)

Ammo same as Above, Instant Rearm with Vanilla, sort of Infinite Ammo without reloads for Vehicles with Large Magazines that don't reload after one shot.

With Ace it's a lot more harmless and less Cheesy but any Ammo Vehicle allows Editing of Pylons so thats back on the Menu for currently 500€.


Medical Tent seems ok, with Vanilla you can just full heal yourself, with Ace and if its enabled via Settings you can use it for Medical Training boost to get downed people up easier/Fix the Medics.
Also it being big, clumsy and not good cover makes the current price pretty reasonable.

**Review by Lazejun — CHANGES_REQUESTED** [review]
Most seems ok, Stuff works except Packing, see Comment.

We probably have to talk about the Prices so that it's not constantly available/useable early Game

File: A3A/addons/logistics/Cargo/Vanilla.hpp [hunk 2]
```diff
@@ -163,3 +163,24 @@ class A3_Supplies_F_Heli_Slingload_Slingload_01_Fuel_F_p3d : TRIPLES(ADDON,Cargo
...
+    offset[] = {0,0,0.65};
+    rotation[] = {0,1,0};
+    size = 2;
+};
+
+class A3_Structures_F_Heli_Ind_Cargo_Cargo10_military_green_F_p3d: TRIPLES(ADDON,Cargo,Base)
+{
+    offset[] = {0,0.1,1.35};
+    rotation[] = {1,0,0};
+    size = 4;
```
[the remark this answers is not in the document]
Comment (killerswin2) [reply A3A/addons/logistics/Cargo/Vanilla.hpp]: increased the size too 5

**Review by Lazejun — APPROVED** [review]

**Comment by Bob-Murphy** [comment]
Please add support which enables us to not have any of the added options for cases in which the modsets/templates either do not have matching assets and the vanilla ones can not be utilized.

**Comment by killerswin2** [comment]
already implemented in the code. If one of the templates defines it as "" then it is skip as that's not a valid cfgVechicle class.

File: A3A/addons/garage/Public/fn_callbackHandler.sqf [hunk 3]
```diff
@@ -91,5 +91,71 @@ switch _callBackName do {
...
+                _arguments params  [
+                    ["_unit", objNull, [objNull]],
+                    ["_spawnItem", "", [""]],       // easier to keep the format from the buyitem then writing a new one
+                    ["_price", -1, [0]],
+                    ["_callbacks", [], [[]]],
+                    ["_object", objNull, [objNull]]
+                ];
+
+                private _lastTimePurchase = _unit getVariable["A3A_spawnItem_cooldown",time];
+                if (_lastTimePurchase > time) exitwith {["Item Purchase", format ["You already bought one, wait %1 seconds before you can buy another.", ceil (_lastTimePurchase - time)]] call A3A_fnc_customHint; deleteVehicle _vehicle;};
```
Comment (Bob-Murphy) [inline A3A/addons/garage/Public/fn_callbackHandler.sqf:108]: Localize please.

File: A3A/addons/garage/Public/fn_callbackHandler.sqf [hunk 4]
```diff
@@ -91,5 +91,71 @@ switch _callBackName do {
...
+                            [0,(-_price)] remoteExec ["A3A_fnc_resourcesFIA",2];
+                            true;
+                        } else {
+                            if ((_unit getVariable ["moneyX", 0]) >= _price) then {
+                                [-_price] call A3A_fnc_resourcesPlayer;
+                                true;
+                            };
+                        };
+                    };
+                    if (_insufficientFunds) exitwith {["Item Purchase", "You can't afford this Item."] call A3A_fnc_customHint};
```
Comment (Bob-Murphy) [inline A3A/addons/garage/Public/fn_callbackHandler.sqf:123]: Localize please.

File: A3A/addons/logistics/Private/fn_unpackObject.sqf [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+//get data 
+private _price = _object getVariable ["A3A_itemPrice", 0];
+private _canOpenDoors = _object getVariable ["A3A_canOpenDoor", false];
+private _itemClassName = _object getVariable ["A3A_packedObject", ""];
+
+// refund if error
+if(_itemClassName isEqualTo "") exitwith {
+    ["Packing Objects", "refunding money to do error"] call A3A_fnc_customHint;
```
Comment (Bob-Murphy) [inline A3A/addons/logistics/Private/fn_unpackObject.sqf:31]: Localize please.

File: A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf [hunk 6]
```diff
@@ -31,32 +31,33 @@ if (!canSuspend) exitwith{};
 if (!hasInterface) exitwith{};
 if (isNull _unit) exitwith {};
 if (!isClass (configFile/"CfgVehicles"/_spawnItem)) exitwith {};
-if (_price == 0) exitwith {};
+if (_price == -1) exitwith {};
 
 //check to make sure that the player is not spamming
 private _lastTimePurchase = _unit getVariable["A3A_spawnItem_cooldown",time];
 if (_lastTimePurchase > time) exitwith {["Item Purchase", format ["You already bought one, wait %1 seconds before you can buy another.", ceil (_lastTimePurchase - time)]] call A3A_fnc_customHint;};
```
Comment (Bob-Murphy) [inline A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf:38]: Localize please.

File: A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf [hunk 7]
```diff
@@ -31,32 +31,33 @@ if (!canSuspend) exitwith{};
...
+            [0,(-_price)] remoteExec ["A3A_fnc_resourcesFIA",2];
             true;
+        } else {
+            if ((_unit getVariable ["moneyX", 0]) >= _price) then {
+                [-_price] call A3A_fnc_resourcesPlayer;
+                true;
+            };
         };
     };
+    if (_insufficientFunds) exitwith {["Item Purchase", "You can't afford this Item."] call A3A_fnc_customHint};
```
Comment (Bob-Murphy) [inline A3A/addons/core/functions/UtilityItems/fn_buyItem.sqf:53]: Localize please.

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]
Sorry, found more small shit to complain about :D

File: A3A/addons/logistics/Private/fn_unpackObject.sqf (hunk 5 above)
Comment (Bob-Murphy) [reply A3A/addons/logistics/Private/fn_unpackObject.sqf:31]: also, what?
`refunding money to do error`

Comment (Bob-Murphy) [reply A3A/addons/logistics/Private/fn_unpackObject.sqf:31]: Oooooh, do you mean `due to error`?

Comment (killerswin2) [reply A3A/addons/logistics/Private/fn_unpackObject.sqf:31]: yes

**Pull request merged**